website/client/js/services/costumeServices.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+(function(){
+  angular
+    .module('habitrpg')
+    .factory('Costume', costumeFactory);
+
+  costumeFactory.$inject = [
+    'Content'
+  ];
+
+  function costumeFactory(Content) {
+    
+    function formatAnimal(name, type) {
+      if(type === 'pet' && name in Content.petsInfo) {
+        return Content.petsInfo[name].text();
+      } else if(type === 'mount' && name in Content.mountsInfo) {
+        return Content.mountsInfo[name].text();
+      }
+    }
+
+    function bgString(set, bg) {
+      var setString = window.env.t(set);
```

Comment:
That would make things easier 😄
On Mon, Sep 12, 2016 at 9:39 PM Blade Barringer <notifications@github.com>
wrote:

> In website/client/js/services/costumeServices.js
> <https://github.com/HabitRPG/habitrpg/pull/7768#discussion_r78488465>:
>
> > +  costumeFactory.$inject = [
> > +    'Content'
> > +  ];
> > +
> > +  function costumeFactory(Content) {
> > +
> > +    function formatAnimal(name, type) {
> > +      if(type === 'pet' && name in Content.petsInfo) {
> > +        return Content.petsInfo[name].text();
> > +      } else if(type === 'mount' && name in Content.mountsInfo) {
> > +        return Content.mountsInfo[name].text();
> > +      }
> > +    }
> > +
> > +    function bgString(set, bg) {
> > +      var setString = window.env.t(set);
>
> I feel like this logic is pretty messy. We can probably do something in
> the content files to make this easier to reason. The content.backgrounds
> object is organized by set, but I think we could just do
> content.backgroundsFlat that just had each background at the root of the
> object.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/7768/files/72a6d73a9be23fd72ba98fbe74b2f14fbc840bfb#r78488465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALz_8yB-N91x1W_SIzBo9HKFQSqorEPNks5qpgzUgaJpZM4JIjd_>
> .
>